### PR TITLE
[14.0][FW] web_m2x_options: multiple ports from 12.0, 13.0

### PR DESCRIPTION
**Pull request by nguyenhk99**
Port of the following PRs from 12.0, 13.0 to 14.0:
* #1570 
* #1500 
* #1728

### CODE REVIEW

File: web_m2x_options/static/src/js/form.js [hunk 1]
```diff
@@ -479,4 +475,21 @@ odoo.define("web_m2x_options.web_m2x_options", function (require) {
...
+    // system parameter "web_m2x_options.field_limit_entries".
+    FormView.include({
+        _setSubViewLimit: function (attrs) {
+            this._super(attrs);
+            var limit = 0;
+            ir_options.forEach((element) => {
+                if (element.key == "web_m2x_options.field_limit_entries") {
+                    limit = element.value;
+                }
+            });
```
Comment (hailangvn) [inline web_m2x_options/static/src/js/form.js]: May I know why this is different than original? https://github.com/OCA/web/pull/1728/commits/a9c15b35ec54ab7ca2da3f75e413bb553690e8c7#diff-d19550cf74a7a0c677bde2f3bf36f00cabf3b3e8b2518402ce2fd53e7e329b87R370

Comment (nguyenhk99) [reply web_m2x_options/static/src/js/form.js]: Hi @hailangvn , `ir_options["web_m2x_options.field_limit_entries"]` return `Array`. Each value in `Array` is an `Object` so I have to use loop to get `element.value`. Do you have a better idea?
image

**Comment by nguyenhk99** [comment]
I'm checking the `ir_options`.

**Review by sebalix — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Can you please do the same for 15.0?

**Pull request merged**